contracts/strategy/TradeFintechStrategy.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright Fathom 2024
+pragma solidity 0.8.19;
+
+import "./BaseStrategy.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+
+// solhint-disable
+contract TradeFintechStrategy is BaseStrategy {
+    using SafeERC20 for ERC20;
+    using Math for uint256;
+
+    uint256 public totalInvested;
+    uint256 public totalGains;
+    uint256 public totalLosses;
+    uint256 public depositLimit;
+
+    // Define periods
+    uint256 public depositPeriodEnds;
+    uint256 public lockPeriodEnds;
+
+    event FundsReturned(address indexed sender, uint256 amount);
+    event GainReported(address indexed sender, uint256 gain);
+    event LossReported(address indexed sender, uint256 loss);
+
+    error ZeroValue();
+    error InvalidPeriods();
+
+    constructor(
+        address _asset,
+        string memory _name,
+        address _tokenizedStrategyAddress,
+        uint256 _depositPeriodEnds,
+        uint256 _lockPeriodEnds,
+        uint256 _depositLimit
+    ) BaseStrategy(_asset, _name, _tokenizedStrategyAddress) {
+        if (_depositPeriodEnds < block.timestamp || _lockPeriodEnds < block.timestamp) {
+            revert InvalidPeriods();
+        }
+        depositPeriodEnds = _depositPeriodEnds;
+        lockPeriodEnds = _lockPeriodEnds;
+        depositLimit = _depositLimit;
+    }
+
+    function _harvestAndReport() internal view override returns (uint256 _totalAssets) {
+        _totalAssets = totalInvested + asset.balanceOf(address(this));
+    }
+
+    function availableDepositLimit(address /*_owner*/) public view override returns (uint256) {
+        // Return the remaining room.
+        return depositLimit - asset.balanceOf(address(this)) > 0 ? depositLimit - asset.balanceOf(address(this)) : 0;
```

Comment:
We will have an underflow error here in case` asset.balanceOf(address(this)) > depositLimit`